src/_pytest/recwarn.py

```diff
@@ -108,8 +108,7 @@ def warns(  # noqa: F811
         ...     warnings.warn("this is not here", UserWarning)
         Traceback (most recent call last):
           ...
-        Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
-
+        _pytest.outcomes.Failed: DID NOT WARN. No warnings of type ...UserWarning... was emitted...
```

Comment:
I think that's fine - and is in line what we get in other places with py3 compared to py2.

Reply:
i do wonder if we want a follow-up to wrap the python trace-back formatting to ensure consistent display (with a way to opt out)

Reply:
I guess in this case it is more related to doctest / Python internals, not the traceback module (https://github.com/python/cpython/blob/24ddd9c2d6ab61cbce7e68d6de36d4df9bd2c3fb/Lib/traceback.py#L567-L568).

Reply:
i do wonder if we should perhaps take a look at setting it to `pytest` instead of leaving it at the internal name

Reply:
Makes sense.  I guess we could set `__qualname__` then.
But it should be the importable name, i.e. `pytest.fail.Exception` then.
Should I do that?

Reply:
-1 on that

the things we do expose on pytest, can have pytest, the  ones hidden stay hidden for now

Reply:
Ok, what do you suggest then?
Keep the hack as is?
Or remove it via this PR (unchanged)?

Reply:
upon reading on the computer i believe i misunderstood

i was suggesting to use a importable name, i didn't take into account that currently we only expose them on the functions